src/main/kotlin/org/rust/ide/highlight/RsHighlightExitPointsHandlerFactory.kt

```diff
@@ -0,0 +1,97 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.highlight
+
+import com.intellij.codeInsight.highlighting.HighlightUsagesHandlerBase
+import com.intellij.codeInsight.highlighting.HighlightUsagesHandlerFactoryBase
+import com.intellij.openapi.editor.Editor
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiFile
+import com.intellij.util.Consumer
+import org.rust.lang.core.psi.*
+import org.rust.lang.core.psi.RsElementTypes.*
+import org.rust.lang.core.psi.ext.elementType
+import org.rust.lang.core.psi.ext.parentOfType
+
+class RsHighlightExitPointsHandlerFactory : HighlightUsagesHandlerFactoryBase() {
+    override fun createHighlightUsagesHandler(editor: Editor, file: PsiFile, target: PsiElement): HighlightUsagesHandlerBase<*>? {
+        if (file !is RsFile) {
+            return null
+        }
+        val elementType = target.elementType
+        if (elementType == RETURN || elementType == TRY_EXPR || elementType == Q) {
+            return RsHighlightExitPointsHandler(editor, file, target)
+        }
+        if (elementType == IDENTIFIER) {
+            val parent = target.parent
+            if (parent is RsMacroCall && (parent.referenceName == "panic" || parent.referenceName == "try")) {
+                return RsHighlightExitPointsHandler(editor, file, target)
+            }
+        }
+        if (testForExprBeforeBlock(target)) {
+            return RsHighlightExitPointsHandler(editor, file, target)
+        }
+        return null
+    }
+     private fun testForExprBeforeBlock(target: PsiElement) : Boolean {
+         var context = target
```

Comment:
You need to specify the `context`'s type (`PsiElement?`) explicitly here. Otherwise, it's inferred as non-nullable `PsiElement` and NPE is thrown when `context = context.parent` hits `null`.